pkg/providers/pricing/pricing.go

```diff
@@ -135,7 +136,9 @@ func (p *Provider) SpotPrice(instanceType string, zone string) (float64, bool) {
 			return val.defaultPrice, true
 		}
 		if price, ok := p.spotPrices[instanceType].prices[zone]; ok {
-			return price, true
+			if onDemandPrice, onDemandOk := p.onDemandPrices[instanceType]; (onDemandOk && price <= onDemandPrice) || !onDemandOk {
```

Comment:
As I understand, this, in the current code, this should always return true, since the onDemand price will always be available if the spot price is available, and with no discounts, this is the guarantee of spot vs on-demand (as there'd be no benefit to spot if on-demand was cheaper).

Reply:
Yes it would always be true in the case with no discounts to lower the on-demand pricing